usecase/src/main/java/org/mobilitydata/gtfsvalidator/usecase/ValidateAllOptionalFileName.java

```diff
@@ -0,0 +1,40 @@
+package org.mobilitydata.gtfsvalidator.usecase;
+
+import org.mobilitydata.gtfsvalidator.usecase.notice.OptionalFileFoundNotice;
+import org.mobilitydata.gtfsvalidator.usecase.port.GtfsSpecRepository;
+import org.mobilitydata.gtfsvalidator.usecase.port.RawFileRepository;
+import org.mobilitydata.gtfsvalidator.usecase.port.ValidationResultRepository;
+
+import java.util.List;
+
+public class ValidateAllOptionalFileName {
+
+    private final GtfsSpecRepository specRepo;
+    private final RawFileRepository rawFileRepo;
+    private final ValidationResultRepository resultRepo;
+    private final List<String> filenameList;
```

Comment:
You don't need to include the type in the variable name - this can be `filenames`

Reply:
That was done at my express request. From my experience `s` postfixes can lead to code where singular and plural forms coexist, sometimes leading to hard to debug issue because of a bad autocomplete. Hence explicit postfix. I'm quite open to discussion on that one.